Code/Framework/AzToolsFramework/Tests/Prefab/PrefabUndoComponentPropertyOverrideTests.cpp

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Prefab/PrefabUndoComponentPropertyTestFixture.h>
+
+#include <AzToolsFramework/Prefab/Undo/PrefabUndoComponentPropertyOverride.h>
+
+namespace UnitTest
+{
+    using PrefabUndoComponentPropertyOverrideTests = PrefabUndoComponentPropertyTestFixture;
+
+    TEST_F(PrefabUndoComponentPropertyOverrideTests, EditTransformStaticSucceeds)
+    {
+        InstanceInfo carInstanceInfo;
+        EntityInfo wheelEntityInfo;
+        ASSERT_TRUE(CreateCarPrefabHierarchy(carInstanceInfo, wheelEntityInfo));
+
+        const bool defaultStaticValue = false;
+        const bool overriddenStaticValue = true;
+
+        // Modify the IsStatic property in transform component as override
+        // Note: Entity needs to be modified manually since we would update cached instance DOM in undo node to avoid reloading
+        AZ::TransformBus::Event(wheelEntityInfo.m_entityId, &AZ::TransformInterface::SetIsStaticTransform, overriddenStaticValue);
+
+        PropertyChangePatch changePatch = MakeTransformStaticPropertyChangePatch(wheelEntityInfo.m_entityId, overriddenStaticValue);
+
+        PrefabDom propertyDomValue;
+        ConvertToPrefabDomValue(propertyDomValue, changePatch.m_propertyValue);
+
+        AZStd::string wheelEntityAliasPath = m_instanceToTemplateInterface->GenerateEntityAliasPath(wheelEntityInfo.m_entityId);
+        AZ::Dom::Path pathToPropertyFromOwningPrefab = AZ::Dom::Path(wheelEntityAliasPath) / changePatch.m_pathToProperty;
+
+        // Create an undo node
+        PrefabUndoComponentPropertyOverride undoNode("Modify transform static");
+
+        InstanceOptionalReference carInstance = m_instanceEntityMapperInterface->FindOwningInstance(carInstanceInfo.m_containerEntityId);
+        EXPECT_TRUE(carInstance.has_value());
+
+        // Redo
+        undoNode.CaptureAndRedo(carInstance->get(), pathToPropertyFromOwningPrefab, propertyDomValue);
+        EXPECT_TRUE(undoNode.Changed());
+        PropagateAllTemplateChanges();
+
+        bool currentStaticValue = defaultStaticValue; // default to opposite value
+        AZ::TransformBus::EventResult(currentStaticValue, wheelEntityInfo.m_entityId, &AZ::TransformBus::Events::IsStaticTransform);
```

Comment:
We could add a check here to validate that wheelEntityInfo.m_entityId is still a valid entity and wasn't recreated from the PropagateAllTemplateChanges().

Reply:
I think we're certain that (like what we're currently doing in other tests) the entity id is valid and unchanged after PropagateAllTemplateChanges(). But the entity object needs to be retrieved again. In these tests, the following calls use the entity id rather than the entity reference, so I think it is okay.

Reply:
I guess if the entity gets recreated and entityId changed, then the test would fail anyway in the next line because the expected transform value would be wrong.